src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -252,10 +252,12 @@ snc_quota_manager::calc_node_quota_default() const {
 
 void snc_quota_manager::get_or_create_quota_context(
   std::unique_ptr<snc_quota_context>& ctx,
-  std::optional<std::string_view> client_id) {
+  std::optional<std::string_view> client_id,
+  const ss::net::inet_address& client_addr,
+  const uint16_t client_port) {
     if (likely(ctx)) {
         // NB: comparing string_view to sstring might be suboptimal
-        if (likely(ctx->_client_id == client_id)) {
+        if (likely(principal_match && ctx->_client_id == client_id)) {
```

Comment:
does the string view get promoted to string, or the string promoted to string_view? maybe explicit `ss::string_view/cast` ? curious.

Reply:
> does the string view get promoted to string, or the string promoted to string_view

The latter. 

`ss::sstring` is on the lefthand side, and its only `operator==` is `operator==(const basic_sstring&)`. For that to be used here, `string_view` on the right needs to be converted to `sstring` but there is no way to do it in one hop (`sstring`'s convestion ctor from `string_view` is explicit, so it would have been `string_view`->`string`->`sstring`).

Actually thanks for asking, now it's clear that performance is ok here. I will remove the comment.

> maybe explicit ss::string_view/cast ?

that would be hard because both sides are under `std::optional`. Waiting eagerly for [std::optional::transform](https://en.cppreference.com/w/cpp/utility/optional/transform).